### ISSUE TITLE
Vs code insider installer hangs forever at "Extracting Files"

### ISSUE THREAD

**Issue by flobo09** [microsoft/vscode#231657]
Does this issue occur when all extensions are disabled?: Yes / NA

- VS Code Version:  Insider x64-1.95.0
- OS Version:  27729.1000

Steps to Reproduce:

1. Launch the insider installer, click forward a wait. 
2. Installer hangs at "Extracting Files". 

The whole program is actually extracted & working, just not registered anywhere in Windows. 
![Image](https://github.com/user-attachments/assets/80eee0b4-2325-42da-bd2c-cf28c153cfa4)

Issue does NOT appear in non insider release which installs correctly. 

I do try it every few days to see if a new release & windows build fixed something but so far it has not happened.

**Comment by flobo09** [comment]
Don't hesitate to ping me if you need logs / test. I looked myself in event viewer but could not see anything. I don't know much about electron apps though so maybe i'm not looking at the right place.

**Comment by l-33ter** [comment]
![Image](https://github.com/user-attachments/assets/e5fbc90d-37c6-464c-ad04-5843e700d2c5) - Same issue. Win11 Insider (Build: 27729)

**Comment by mcsimw** [comment]
![Image](https://github.com/user-attachments/assets/fb423eda-49c4-415e-9d09-5458ec96037f)

Same issue:
VS Code Version: Insider x64-1.95.0
OS Version: 27729.1000

**Comment by mcsimw** [comment]
issue persists on VS Code Version: Insider x64-1.96.0

**Comment by RuiRui369** [comment]
您好，我是一名初学者，第一次接触vscode insiders，我安装1.96.0版本到最后，也是一直在解压缩文件，应该是卡死了，请问，这是怎么回事呢？自我下载到现在已经有3天了，稳定版本我可以正常使用，但insiders版本仍旧是这样，难道是我的系统有问题，还是这个有BUG？

Hello, I'm a beginner, the first contact with vs code insiders, I installed version 1.96.0 to the end, and I've been unzipping the file, it should be stuck, excuse me, what's going on? It's been 3 days since I downloaded it. The stable version I can use normally. But the insiders version is still like this. Is there something wrong with my system. Or is this a bug?

https://github.com/user-attachments/assets/07b17cfe-1012-465b-b832-7b7a226e56ed

**Comment by RuiRui369** [comment]
您好，我是一名初学者，第一次接触vscode insiders，我安装1.96.0版本到最后，也是一直在解压缩文件，应该是卡死了，请问，这是怎么回事呢？自我下载到现在已经有3天了，稳定版本我可以正常使用，但insiders版本仍旧是这样，难道是我的系统有问题，还是这个有BUG？可以肯定的是，就像您说的那样，可以从已下载的文件夹中找到应用程序并且使用没有问题，但是在Windows中无法找到该文件，应该是并没有注册进去，这个问题让我这个完美主义者十分头疼，不知道各位编程大佬有何办法。。。

With machine translation : 
Hello, I'm a beginner, the first contact with vs code insiders, I installed version 1.96.0 to the end, and I've been unzipping the file, it should be stuck, excuse me, what's going on? It's been 3 days since I downloaded it. The stable version I can use normally. But the insiders version is still like this. Is there something wrong with my system. Or is this a bug? To be sure, as you said, you can find the app from the downloaded folder and use it without problems, but you can't find the file in Windows, so it shouldn't be registered, which is a headache for me as a perfectionist, and I don't know what you programming guys can do.

https://github.com/user-attachments/assets/a299cd1d-80ad-47a7-ae31-90ba48210da0

![Image](https://github.com/user-attachments/assets/8da3f73f-f68c-4033-8922-b376ed01e565)

**Comment by PeripheralVisionPD2** [comment]
Figured out how to fix this. Once it hangs, go into Task Manager and kill all PowerShell instances that show up, it should then complete the install successfully.

**Comment by RuiRui369** [comment]
> Figured out how to fix this. Once it hangs, go into Task Manager and kill all PowerShell instances that show up, it should then complete the install successfully.

This installation problem should be related to the latest preview version of win11 system, when I installed it, there were no PowerShell instances running on the Task Manager. One solution is to roll back to the stable version of win11. My friend with 24H2 can install it completely.

**Comment by langningchen** [comment]
> Figured out how to fix this. Once it hangs, go into Task Manager and kill all PowerShell instances that show up, it should then complete the install successfully.

It works, thank you!

**Comment by ChattyEnol** [comment]
Yeah! I met the same issue.

One probable way to solve it may be reboot your PC to `Security Mode` , and re-run this installer. I installed VS Code Insiders successfully in this way.

But, there is still something wrong with VS Code Insiders. It cannot be updated...

**Comment by Shahbazian-Will** [comment]
> Yeah! I met the same issue.
> 
> One probable way to solve it may be reboot your PC to `Security Mode` , and re-run this installer. I installed VS Code Insiders successfully in this way.
> 
> But, there is still something wrong with VS Code Insiders. It cannot be updated...

Mine seems to be hanging on "installing update", is this what is happening to you?

**Comment by ChattyEnol** [comment]
> > Yeah! I met the same issue.
> > 
> > One probable way to solve it may be reboot your PC to `Security Mode` , and re-run this installer. I installed VS Code Insiders successfully in this way.
> > 
> > But, there is still something wrong with VS Code Insiders. It cannot be updated...
> 
> Mine seems to be hanging on "installing update", is this what is happening to you?

Yes. So I just download the system installer of VS Code Insiders on the official website of VS Code:

https://code.visualstudio.com/insiders/#win

and reboot my PC to Security Mode and run the installer I downloaded. I update my VS Code Insiders manually in this way😂

**Comment by RuiRui369** [comment]
> > > Yeah! I met the same issue.
> > > One probable way to solve it may be reboot your PC to `Security Mode` , and re-run this installer. I installed VS Code Insiders successfully in this way.
> > > But, there is still something wrong with VS Code Insiders. It cannot be updated...
> > 
> > 
> > Mine seems to be hanging on "installing update", is this what is happening to you?
> 
> Yes. So I just download the system installer of VS Code Insiders on the official website of VS Code:
> 
> https://code.visualstudio.com/insiders/#win
> 
> and reboot my PC to Security Mode and run the installer I downloaded. I update my VS Code Insiders manually in this way😂

启用安全模式也许真的可以解决这个应用安装最后的卡死，但是问题在于就像你为了解决AutoCAD的一系列问题花时间重装系统一样，既不能更好的完成其他工作，也浪费了大量时间，安全模式下好多地方不方便。就目前我的测试，通过后台关闭已经安装完毕但是未弹出成功界面的insiders版本，就可以下载到电脑上，只是电脑无法识别，同时当insiders版有更新时再点击更新按钮仍会卡死，但是使用等方面没有太大影响，所以还是等工程师把这完全修好吧，挺令人头疼的。。。

Enabling safe mode may really solve the problem of the app getting stuck during installation, but the problem is that just like how you spend time reinstalling the system to solve a series of problems with AutoCAD, it cannot better complete other tasks and wastes a lot of time. Safe mode is inconvenient in many places. Based on my current testing, the Insiders version, which has been installed but did not pop up successfully, can be downloaded to the computer by closing it in the background. However, the computer cannot recognize it, and when the Insiders version is updated, clicking the update button will still cause it to freeze. However, there is not much impact on usage and other aspects, so it is better to wait for the engineer to completely fix this, which is quite troublesome...

**Comment by ChattyEnol** [comment]
> > > > Yeah! I met the same issue.
> > > > One probable way to solve it may be reboot your PC to `Security Mode` , and re-run this installer. I installed VS Code Insiders successfully in this way.
> > > > But, there is still something wrong with VS Code Insiders. It cannot be updated...
> > > 
> > > 
> > > Mine seems to be hanging on "installing update", is this what is happening to you?
> > 
> > Yes. So I just download the system installer of VS Code Insiders on the official website of VS Code:
> > 
> > https://code.visualstudio.com/insiders/#win
> > 
> > and reboot my PC to Security Mode and run the installer I downloaded. I update my VS Code Insiders manually in this way😂
> 
> 启用安全模式也许真的可以解决这个应用安装最后的卡死，但是问题在于就像你为了解决AutoCAD的一系列问题花时间重装系统一样，既不能更好的完成其他工作，也浪费了大量时间，安全模式下好多地方不方便。就目前我的测试，通过后台关闭已经安装完毕但是未弹出成功界面的insiders版本，就可以下载到电脑上，只是电脑无法识别，同时当insiders版有更新时再点击更新按钮仍会卡死，但是使用等方面没有太大影响，所以还是等工程师把这完全修好吧，挺令人头疼的。。。
> 
> Enabling safe mode may really solve the problem of the app getting stuck during installation, but the problem is that just like how you spend time reinstalling the system to solve a series of problems with AutoCAD, it cannot better complete other tasks and wastes a lot of time. Safe mode is inconvenient in many places. Based on my current testing, the Insiders version, which has been installed but did not pop up successfully, can be downloaded to the computer by closing it in the background. However, the computer cannot recognize it, and when the Insiders version is updated, clicking the update button will still cause it to freeze. However, there is not much impact on usage and other aspects, so it is better to wait for the engineer to completely fix this, which is quite troublesome...

我受不了了，重装了，重装后没问题😂

**Comment by RuiRui369** [comment]
> > > > > Yeah! I met the same issue.
> > > > > One probable way to solve it may be reboot your PC to `Security Mode` , and re-run this installer. I installed VS Code Insiders successfully in this way.
> > > > > But, there is still something wrong with VS Code Insiders. It cannot be updated...
> > > > 
> > > > 
> > > > Mine seems to be hanging on "installing update", is this what is happening to you?
> > > 
> > > 
> > > Yes. So I just download the system installer of VS Code Insiders on the official website of VS Code:
> > > https://code.visualstudio.com/insiders/#win
> > > and reboot my PC to Security Mode and run the installer I downloaded. I update my VS Code Insiders manually in this way😂
> > 
> > 
> > 启用安全模式也许真的可以解决这个应用安装最后的卡死，但是问题在于就像你为了解决AutoCAD的一系列问题花时间重装系统一样，既不能更好的完成其他工作，也浪费了大量时间，安全模式下好多地方不方便。就目前我的测试，通过后台关闭已经安装完毕但是未弹出成功界面的insiders版本，就可以下载到电脑上，只是电脑无法识别，同时当insiders版有更新时再点击更新按钮仍会卡死，但是使用等方面没有太大影响，所以还是等工程师把这完全修好吧，挺令人头疼的。。。
> > Enabling safe mode may really solve the problem of the app getting stuck during installation, but the problem is that just like how you spend time reinstalling the system to solve a series of problems with AutoCAD, it cannot better complete other tasks and wastes a lot of time. Safe mode is inconvenient in many places. Based on my current testing, the Insiders version, which has been installed but did not pop up successfully, can be downloaded to the computer by closing it in the background. However, the computer cannot recognize it, and when the Insiders version is updated, clicking the update button will still cause it to freeze. However, there is not much impact on usage and other aspects, so it is better to wait for the engineer to completely fix this, which is quite troublesome...
> 
> 我受不了了，重装了，重装后没问题😂

哈哈哈，受不了你了，这是强迫症的噩梦啊。。。😂😂😂
hahaha , It's so fun. It's an obsessive-compulsive nightmare.😂😂😂

**Comment by CanePlayz** [comment]
Updates work again for me with the newest Windows Insider Canary build

**Comment by DefamationStation** [comment]
> Figured out how to fix this. Once it hangs, go into Task Manager and kill all PowerShell instances that show up, it should then complete the install successfully.

absolute lifesaver. 1.98 insiders and this still happens on a fresh system

**Comment by RuiRui369** [comment]
> > Figured out how to fix this. Once it hangs, go into Task Manager and kill all PowerShell instances that show up, it should then complete the install successfully.
> 
> absolute lifesaver. 1.98 insiders and this still happens on a fresh system

I guess Microsoft has already given up on this bug , Just let the bug continue to exist. Hahaha ... It's been almost 4 months and it hasn't been fixed yet ...

**Comment by PieterVerledens** [comment]
Possible duplicate issues: #234016 #239521 #237665 #231526 #239329 #231436.

**Comment by mcsimw** [comment]
> > Figured out how to fix this. Once it hangs, go into Task Manager and kill all PowerShell instances that show up, it should then complete the install successfully.
> 
> absolute lifesaver. 1.98 insiders and this still happens on a fresh system

Terminating the PowerShell process will allow the VSCode Insiders installation to proceed, but any settings configured during the installation, such as adding VSCode to context menus, will not be applied.

**Comment by mrhyde** [comment]
PowerShell process termination fails with error (see screenshot). According to the logs, VSCode attempts to find the installation in the incorrect directory. My VSCode Insiders is installed in `%AppData%\Local\Programs\Microsoft VS Code Insiders` instead of the path shown in the error logs and I have never installed a system version of VS Code.

![Image](https://github.com/user-attachments/assets/32bc3534-ba2a-4653-8171-4f4a50d72355)

[vscode-inno-updater-1740102811.log](https://github.com/user-attachments/files/18897756/vscode-inno-updater-1740102811.log)

Version: 1.97.0-insider
Commit: 33fc5a94a3f99ebe7087e8fe79fbe1d37a251016
Date: 2025-02-04T18:50:12.241Z (2 wks ago)
Electron: 32.2.7
ElectronBuildId: 10660205
Chromium: 128.0.6613.186
Node.js: 20.18.1
V8: 12.8.374.38-electron.0
OS: Windows_NT x64 10.0.27788


<details>
  <summary> tree C:\Program Files\Microsoft VS Code Insiders -L 2</summary>

```sh
├── _
│   ├── bin
│   ├── locales
│   ├── policies
│   ├── resources
│   ├── chrome_100_percent.pak
│   ├── chrome_200_percent.pak
│   ├── Code - Insiders.exe
│   ├── Code - Insiders.VisualElementsManifest.xml
│   ├── d3dcompiler_47.dll
│   ├── debug.log
│   ├── ffmpeg.dll
│   ├── icudtl.dat
│   ├── libEGL.dll
│   ├── libGLESv2.dll
│   ├── LICENSES.chromium.html
│   ├── resources.pak
│   ├── snapshot_blob.bin
│   ├── v8_context_snapshot.bin
│   ├── vk_swiftshader.dll
│   ├── vk_swiftshader_icd.json
│   └── vulkan-1.dll
├── appx
│   ├── code_insiders_explorer_command.dll
│   └── code_insiders_explorer_x64.appx
├── tools
│   ├── inno_updater.exe
│   └── vcruntime140.dll
├── unins000.dat
├── unins000.exe
├── unins001.dat
├── unins001.exe
├── unins002.dat
├── unins002.exe
├── unins003.dat
├── unins003.exe
└── unins003.msg
```

</details>



<details>
  <summary>tree %AppData%\Local\Programs\Microsoft VS Code Insiders -L 2</summary>

```sh
.
├── appx
│   ├── code_insiders_explorer_command.dll
│   └── code_insiders_explorer_x64.appx
├── bin
│   ├── code-insiders
│   ├── code-insiders.cmd
│   └── code-tunnel-insiders.exe
├── locales
│   ├── af.pak
│   ├── am.pak
│   ├── ar.pak
│   ├── bg.pak
│   ├── bn.pak
│   ├── ca.pak
│   ├── cs.pak
│   ├── da.pak
│   ├── de.pak
│   ├── el.pak
│   ├── en-GB.pak
│   ├── en-US.pak
│   ├── es-419.pak
│   ├── es.pak
│   ├── et.pak
│   ├── fa.pak
│   ├── fil.pak
│   ├── fi.pak
│   ├── fr.pak
│   ├── gu.pak
│   ├── he.pak
│   ├── hi.pak
│   ├── hr.pak
│   ├── hu.pak
│   ├── id.pak
│   ├── it.pak
│   ├── ja.pak
│   ├── kn.pak
│   ├── ko.pak
│   ├── lt.pak
│   ├── lv.pak
│   ├── ml.pak
│   ├── mr.pak
│   ├── ms.pak
│   ├── nb.pak
│   ├── nl.pak
│   ├── pl.pak
│   ├── pt-BR.pak
│   ├── pt-PT.pak
│   ├── ro.pak
│   ├── ru.pak
│   ├── sk.pak
│   ├── sl.pak
│   ├── sr.pak
│   ├── sv.pak
│   ├── sw.pak
│   ├── ta.pak
│   ├── te.pak
│   ├── th.pak
│   ├── tr.pak
│   ├── uk.pak
│   ├── ur.pak
│   ├── vi.pak
│   ├── zh-CN.pak
│   └── zh-TW.pak
├── policies
│   ├── cs-cz
│   ├── de-de
│   ├── en-us
│   ├── es-es
│   ├── fr-fr
│   ├── it-it
│   ├── ja-jp
│   ├── ko-kr
│   ├── pl-pl
│   ├── pt-br
│   ├── ru-ru
│   ├── tr-tr
│   ├── zh-cn
│   ├── zh-tw
│   └── VSCodeInsiders.admx
├── resources
│   └── app
├── tools
│   ├── inno_updater.exe
│   └── vcruntime140.dll
├── chrome_100_percent.pak
├── chrome_200_percent.pak
├── Code - Insiders.exe
├── Code - Insiders.VisualElementsManifest.xml
├── d3dcompiler_47.dll
├── ffmpeg.dll
├── icudtl.dat
├── libEGL.dll
├── libGLESv2.dll
├── LICENSES.chromium.html
├── resources.pak
├── snapshot_blob.bin
├── unins000.dat
├── unins000.exe
├── v8_context_snapshot.bin
├── vk_swiftshader.dll
├── vk_swiftshader_icd.json
└── vulkan-1.dll
```

</details>

**Comment by calvinwonghk** [comment]
yes it works, kill the powershell.exe, thanks! so buggy :D even the installer

**Comment by calvinwonghk** [comment]
the issues still existed on 1.99 installer someone should fix it?

**Comment by scrawnysatyr** [comment]
![Image](https://github.com/user-attachments/assets/1d116892-9a34-4cd7-94ce-57639722a32c)

The installation process completed successfully without any extraction errors or freezes. 

OS Build : Windows 11 Insider Preview Build 27818 (Canary Channel)
SW : Visual Studio Code Insiders 1.99.0 (User)

Due to a failed Windows Update, I did performed a fresh installation of the Windows.

**Comment by Craftomon** [comment]
> Figured out how to fix this. Once it hangs, go into Task Manager and kill all PowerShell instances that show up, it should then complete the install successfully.

What are "PowerShell instances"?

**Comment by Craftomon** [comment]
> > Figured out how to fix this. Once it hangs, go into Task Manager and kill all PowerShell instances that show up, it should then complete the install successfully.
> 
> What are "PowerShell instances"?

Nevermind
